openapi.yaml

```diff
@@ -0,0 +1,1607 @@
+            type: string
+            enum: [B, BI]
+        - name: id
+          in: path
+          required: true
+          description: The UUID of the business or respondent
+          schema:
+            type: string
+            format: uuid
+        - name: survey_id
+          in: query
+          required: false
+          description: The UUID of the survey
+          schema:
+            type: string
+            format: uuid
+        - name: enrolment_status
+          in: query
+          required: false
+          description: The enrolment status
+          schema:
+            type: string
+            enum: [PENDING, ENABLED, DISABLED, SUSPENDED]
+      responses:
+        200:
+          description: The business or respondent has been retrieved
+          content:
+            application/json:
+              schema:
+                oneOf:
+                  - $ref: '#/components/schemas/BusinessResponse'
+                  - $ref: '#/components/schemas/RespondentWithAssociations'
+        400:
+          description: The provided sample unit type wasn't one of B or BI
+        404:
+          description: The business or respondent does not exist
+  /pending-survey-users-count:
+    get:
+      tags:
+        - misc
+      summary: Get the count of users who are already enrolled and pending a survey transfer
+      description: Get the count of a business' users who are already enrolled and pending a survey transfer
+      parameters:
+        - name: business_id
+          in: query
+          required: true
+          description: The business ID to count for
+          schema:
+            type: string
+            format: uuid
+        - name: survey_id
+          in: query
+          required: true
+          description: The survey ID to count for
+          schema:
+            type: string
+            format: uuid
+        - name: is_transfer
+          in: query
+          required: false
+          description: Whether the user should be pending a transfer, rather than a share
+          schema:
+            type: boolean
+      responses:
+        200:
+          description: The count of users has been retrieved
+          content:
+            application/json:
+              schema:
+                type: integer
+        400:
+          description: The request was missing the business or survey ID
+        404:
+          description: The business does not exist
+  /pending-surveys:
+    post:
+      tags:
+        - misc
+      summary: Create a survey transfer or share
+      description: Create a survey transfer or share
+      requestBody:
+        required: true
+        content:
+          application/json:
+            schema:
+              oneOf:
+                - $ref: '#/components/schemas/SurveyShare'
+                - $ref: '#/components/schemas/SurveyTransfer'
+      responses:
+        201:
+          description: The survey transfer or share has been created
+        400:
```

Comment:
pending_shares or pending_transfers not in payload

Reply:
pending_surveys empty as well and then validated against "email_address", "survey_id", "business_id", "shared_by"

Reply:
Respondent missing

Reply:
As above